### ISSUE TITLE
Make test/e2e/scheduler_predicates.go not break once the old docs repo is turned down

### ISSUE THREAD

**Issue by davidopp** [kubernetes/kubernetes#23105]
`test/e2e/scheduler_predicates.go` does this:

```
nodeSelectionRoot := filepath.Join(testContext.RepoRoot, "docs/user-guide/node-selection")
testPodPath := filepath.Join(nodeSelectionRoot, "pod-with-node-affinity.yaml")
runKubectlOrDie("create", "-f", testPodPath, fmt.Sprintf("--namespace=%v", ns))
```

This is a nice pattern -- using config from an example directly in a test, which ensures sure the example stays in sync with the code. But this is going to break once kubernetes/kubernetes/docs/user-guide/ is deleted. I assume what we need to do is to create a `testContext.DocsRepoRoot` variable that points to `kubernetes/kubernetes.github.io` ? It seems that it is currently defined as a relative path (see https://github.com/kubernetes/kubernetes/blob/bf1bb5d30931b8031b929e9c280ee02842c65827/test/e2e/e2e.go#L69 ); it would be nice if we can just define `testContext.DocsRepoRoot` the same way but just append `../kubernetes.github.io/` at the end, but something tells me that won't work...

Any thoughts from the test folks?

cc/ @kubernetes/sig-testing  @kevin-wangzefeng  @johndmulhausen

**Comment by ixdy** [comment]
We don't currently check out the docs repo in CI, so we probably need to do something more sophisticated. Currently this test only works because we [copy `docs/` into the release tarball](https://github.com/kubernetes/kubernetes/blob/master/build/common.sh#L1032). Are we planning to drop the docs from the release tarball?

**Comment by davidopp** [comment]
@kevin-wangzefeng and I discussed this, and we are thinking now maybe a better approach is to move such config files out of docs/ and into examples/. That way they will stay in the main kubernetes repo and the problem described here goes away. (Until we decide to move examples/ into a separate repo...) There is already some precedent, insofar as we have an example for the "scheduler extender" feature in `https://github.com/kubernetes/kubernetes/blob/master/examples/scheduler-policy-config-with-extender.json`

The thing we weren't sure is whether it would still be possible to include these files from docs files. For example, currently
http://kubernetes.io/docs/user-guide/node-selection/
says
```{% include code.html language="yaml" file="pod.yaml" ghlink="/docs/user-guide/node-selection/pod.yaml" %}```
which presumably won't work for referencing a file in another repo (i.e. in kubernetes/examples/). Assuming that won't work cross-repo, we can just put in a link, but @kevin-wangzefeng pointed out that raises the question of what link target to use -- to the current version of the file at HEAD? or in a particular blob, corresponding to the current release?

cc/ @bgrant0607

**Comment by bgrant0607** [comment]
@davidopp @kevin-wangzefeng 

I do not want to move the files to /examples. They don't belong there. There are already many examples that don't belong there, such as scheduler-policy-config-with-extender, which should have been put in the admin guide, with an explanatory doc. Please read examples/README.md.

That would also just push the problem around, since we want to move examples out of the main repo eventually.

The doc move still has lots of loose ends to clean up. This is one of them. The yaml files currently exist in both repos.

The original point of the examples tests was to ensure our docs and examples contained correct manifest files. What we really need to do is to get the examples test working in the doc repo. The problem is that the test links against Kubernetes internal code. The test needs to be completely rewritten to validate via kubectl, which would also resolve the long-standing problem with not being able to deal with multi-object files.

cc @johndmulhausen @kubernetes/docs @kubernetes/examples

**Comment by davidopp** [comment]
To clarify, this isn't about the examples tests that you're referring to. It's about a bespoke end-to-end test that uses an example from the docs. 

Anyway, I'm fine with having the example in the docs repo, but until we fix this issue, we need to be able to merge changes to the old docs repo (currently the bot forbids it), otherwise the new test won't work.

**Comment by bgrant0607** [comment]
@davidopp Are the PRs labeled with do-not-merge? I thought the bot only applied that to PRs touching markdown files.

**Comment by spiffxp** [comment]
what do we think about closing this issue?  this particular test has been addressed on master via #30060 (thanks @gmarek!)

**Comment by spiffxp** [comment]
/sig scheduling
because predicates

/assign
the specific test in question has been fixed for a while, calling this done
